src/components/accessories/patientTherapy/patientTherapyTable/PatientTherapyTable.tsx

```diff
@@ -41,26 +84,42 @@ const PatientTherapyTable: FunctionComponent<IOwnProps> = ({
 
   const onEView = () => {};
 
-  useEffect(() => {
-    setUpdate(shouldUpdateTable);
-  }, [shouldUpdateTable]);
+  const renderSwitch = (status: string = "") => {
+    switch (status) {
+      case "FAIL":
+        return <InfoBox type="error" message={t("common.somethingwrong")} />;
+      case "LOADING":
+        return (
+          <CircularProgress
+            style={{ marginLeft: "50%", position: "relative" }}
+          />
+        );
+
+      case "SUCCESS":
+        return (
+          <Table
+            rowData={formatDataToDisplay(data)}
+            compareRows={dateComparator}
+            tableHeader={header}
+            labelData={label}
+            columnsOrder={order}
+            rowsPerPage={5}
+            onDelete={onDelete}
+            isCollapsabile={true}
+            onEdit={onEdit}
+            onView={onEView}
+          />
+        );
 
+      case "SUCCESS_EMPTY":
+        return <InfoBox type="warning" message={t("common.emptydata")} />;
+
+      default:
+        return;
+    }
+  };
   return (
-    <>
-      <div className="patientTherapyTable">
-        <Table
-          rowData={data}
-          tableHeader={header}
-          labelData={label}
-          columnsOrder={order}
-          rowsPerPage={5}
-          onDelete={onDelete}
-          isCollapsabile={true}
-          onEdit={onEdit}
-          onView={onEView}
-        />
-      </div>
-    </>
+    <div className="patientTherapyTable">{renderSwitch(searchStatus)}</div>
```

Comment:
Same thing here: we don't need a function that renders the switch once we can simply declare it within the brackets.